File: crates/klyron_linter/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron_linter {
LinterBuilder::LinterBuilder() : config_() {}
LinterBuilder& LinterBuilder::enabled(bool v) { config_.enabled = v; return *this; }
LinterBuilder& LinterBuilder::verbose(bool v) { config_.verbose = v; return *this; }
LinterConfig LinterBuilder::build() { return config_; }
}

File: crates/klyron_linter/bindings/rust/builder.rs
use crate::config::LinterConfig;

pub struct LinterBuilder {
    config: LinterConfig,
}

impl LinterBuilder {
    pub fn new() -> Self { Self { config: LinterConfig::default() } }
    pub fn build(self) -> LinterConfig { self.config }
}

impl Default for LinterBuilder {
    fn default() -> Self { Self::new() }
}
